src/config/config_file/network/midnight.rs

```diff
@@ -0,0 +1,341 @@
+//! Midnight Network Configuration
+//!
+//! This module provides configuration support for Midnight blockchain networks.
+
+use super::common::NetworkConfigCommon;
+use crate::config::ConfigFileError;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq, Hash)]
+pub struct IndexerUrls {
+    pub http: String,
+    pub ws: String,
+}
+
+/// Configuration specific to Midnight networks.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct MidnightNetworkConfig {
+    /// Common network fields.
+    #[serde(flatten)]
+    pub common: NetworkConfigCommon,
+    // Midnight-specific fields
+    pub indexer_urls: IndexerUrls, // URL for the indexer server (ws, http)
+    pub prover_url: String,        // URL for the prover server
+    pub commitment_tree_ttl: Option<u64>, // How long to cache Merkle roots
+}
+
+impl MidnightNetworkConfig {
+    /// Validates the specific configuration fields for a Midnight network.
+    ///
+    /// # Returns
+    /// - `Ok(())` if the Midnight configuration is valid.
+    /// - `Err(ConfigFileError)` if validation fails (e.g., missing fields, invalid URLs).
+    pub fn validate(&self) -> Result<(), ConfigFileError> {
+        self.common.validate()?;
+
+        // Validate indexer URLs
+        reqwest::Url::parse(&self.indexer_urls.http).map_err(|_| {
+            ConfigFileError::InvalidFormat(format!(
+                "Invalid indexer HTTP URL: {}",
+                self.indexer_urls.http
+            ))
+        })?;
+
+        reqwest::Url::parse(&self.indexer_urls.ws).map_err(|_| {
+            ConfigFileError::InvalidFormat(format!(
+                "Invalid indexer WebSocket URL: {}",
+                self.indexer_urls.ws
+            ))
+        })?;
+
+        // Validate prover URL if provided
+        reqwest::Url::parse(&self.prover_url).map_err(|_| {
+            ConfigFileError::InvalidFormat(format!("Invalid prover URL: {}", self.prover_url))
+        })?;
+
+        // Validate network_id if provided
+        match self.common.network.as_str() {
```

Comment:
These are correct names?
Should we limit it to testnet given we are not sure how they will call their networks if it is not yet defined?